Update OutputType to WinExe on -windows

Redth opened the pull request:
This happens in the sdk targets/props but not quite early enough to influence the `EnableWin32Codegen` property which causes `[STAThreadAttribute]` to be generated in the Windows app.


Also: For non-blazor projects, interpreter is fine
Blazor android projects (and possibly iOS?) will require this runtime fix: dotnet/runtime#58467
So, enabling interpreter again on non-blazor templates

Commit:
Update OutputType to WinExe on -windows

This happens in the sdk targets/props but not quite early enough to influence the `EnableWin32Codegen` property which causes `[STAThreadAttribute]` to be generated in the Windows app.

mattleibow reviewed (approved):
Might be good to remove the prop from the targets file to avoid confusion.

Commit:
For non-blazor projects, interpreter is fine

Blazor android projects (and possibly iOS?) will require this runtime fix: https://github.com/dotnet/runtime/pull/58467

So, enabling interpreter again on non-blazor templates
